### PR TITLE
Retry for 36 hours on Hutch publish errors

### PR DESCRIPTION
**Pull request by sebbacon**

### CODE REVIEW

File: lib/handler.rb [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
     Hutch.publish(message_type, message)
-  rescue Hutch::PublishError => e
-    if retries > 0
-      sleep 5.0 # to match the default reconnection interval in Bunny
-      Rails.logger.error("#{e.message} (Hutch::PublishError) (retrying) #{message.inspect}")
-      publish_with_retries(message_type, message, retries - 1)
+  rescue Hutch::PublishError, Bunny::ConnectionClosedError => e
+    if interval < MAX_WAIT
+      sleep(interval)
+      Rails.logger.error("#{e.message} (Hutch::PublishError) (attempt #{attempt}, retrying in #{interval * 2}s) #{message.inspect}")
```
Comment (jpmckinney) [inline lib/handler.rb]: `interval * 2` is only equal to `interval * (attempt + 1)` when attempt equals `1`, no? The current code will sleep 1, then 2, then 3, then 4. Not 1, then 2, then 4, then 8.

File: lib/handler.rb [hunk 2]
```diff
@@ -1,13 +1,18 @@
 require 'turbot_runner'
 
-def publish_with_retries(message_type, message, retries = 4)
+MAX_WAIT = 60 * 60 * 24 # 24 hours, amounting to 16 attempts and 36
```
Comment (jpmckinney) [inline lib/handler.rb]: I'm not sure that it'll be only 36 cumulative hours if `attempt` increments by one, and `interval = INITIAL_INTERVAL * attempt`.

File: lib/handler.rb (hunk 1 above)
Comment (sebbacon) [inline lib/handler.rb]: Good spot, it was meant to be `interval = 2**attempt` (cc @inglesp)

**Comment by bensymonds** [comment]
We no longer use rabbitmq/angler so this is dead.

**Pull request closed without merging**